crates/sncast/src/helpers/scarb_utils.rs

```diff
@@ -108,23 +108,31 @@ pub fn get_package_metadata<'a>(
 pub struct BuildConfig {
     pub scarb_toml_path: Utf8PathBuf,
     pub json: bool,
+    pub profile: String,
 }
 
 pub fn build(config: &BuildConfig) -> Result<HashMap<String, StarknetContractArtifacts>> {
     let mut cmd = ScarbCommand::new_with_stdio();
-    cmd.arg("build").manifest_path(&config.scarb_toml_path);
+    let metadata = get_scarb_metadata_command(&config.scarb_toml_path)?
+        .exec()
+        .expect("Failed to obtain metadata");
+    let profile = if metadata.profiles.contains(&config.profile) {
+        &config.profile
+    } else {
+        "dev"
```

Comment:
I suggest we introduce `const SCARB_DEFAULT_PROFILE`

Reply:
why?

Reply:
moved discussion offline, will make any changes in separate PR if necessary